### PR TITLE
Add generated modules that output const &str for `tracing` compatibility

### LINKED ISSUE #1320: Need `&'static str`s from `semantic-conventions` for `tracing` field recording
I'd like to do something like the following with `tracing`:

```rust
event!(
    Level::INFO,
    { semconv::trace::CODE_FUNCTION } = "update_user",
    { semconv::trace::CODE_NAMESPACE } = "com.my-org",
    { my_org::semconv::USER_ID } = request.user_id,
    "Updating user"
);
```

The `tracing` crate requires values that look like [Rust identifiers, string literals, or constant expressions](https://github.com/tokio-rs/tracing/blob/f93cfa087e6ebdcbd8ecdcccca47d73c3a89ab94/tracing/src/lib.rs#L322-L333). But the string for values like code function are wrapped up in a `Key` which although `const` does not satisfy the `format_args` that is used by `tracing`: 

```
error: format argument must be a string literal
  --> sample/src/tracing_test/mod.rs:28:9
   |
28 |         { semconv::trace::CODE_FUNCTION } = "update_user",
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: you might be missing a string literal to format with
   |
28 |         "{} {} {} {}", { semconv::trace::CODE_FUNCTION } = "update_user",
   |         ++++++++++++++
```

I'd think that the easiest solution would be to create another submodule that is code generated with the just primary source of the key strings from which the module with the `const Key`s are derived, e.g.,:

```rust
const CODE_FUNCTION: &str = "code.function";
```

That way something like the above example would work for `tracing` as well as any other context in which `const` / `&'static str` is required.

Although the code generation can be on my end, automatically or manually, this seems basic enough and useful to others that I'd think it should be implemented here? Or is there some Rust technique or library I haven't learned of yet to make use of the existing constants instead of copy and pasting raw `str` values like `"code.function"` everywhere?

### PR DESCRIPTION
**Pull request by kriswuollett**
- Adds generated modules for `const` `&str` access to semantic convention keys, fixes #1320, via `semconv` submodule

### CODE REVIEW

**Review by shaun-cox — APPROVED** [review]
Thanks for the contribution!

File: opentelemetry-semantic-conventions/src/semconv/mod.rs [hunk 1]
```diff
@@ -0,0 +1,2 @@
+pub mod resource;
```
Comment (lalitb) [inline opentelemetry-semantic-conventions/src/semconv/mod.rs:1]: Do we need semconv directory along with the generated files here, now that we are directly modifying the existing generated files ?

Comment (kriswuollett) [reply opentelemetry-semantic-conventions/src/semconv/mod.rs:1]: No we don't need the dir. Removed it my recent force push. Will take off the WIP soon once I double check the tests and other code.

File: opentelemetry-semantic-conventions/src/semconv/resource.rs [hunk 2]
```diff
@@ -0,0 +1,882 @@
+// DO NOT EDIT, this is an auto-generated file
```
Comment (jtescher) [inline opentelemetry-semantic-conventions/src/semconv/resource.rs:1]: Was this file added by mistake? seems to be a dup of src/resource.rs

File: opentelemetry-semantic-conventions/src/semconv/trace.rs [hunk 3]
```diff
@@ -0,0 +1,1599 @@
+// DO NOT EDIT, this is an auto-generated file
```
Comment (jtescher) [inline opentelemetry-semantic-conventions/src/semconv/trace.rs:1]: Same with this one

File: opentelemetry-semantic-conventions/scripts/templates/header_resource.rs [hunk 4]
```diff
@@ -7,14 +7,31 @@
 //!
 //! ## Usage
 //!
+//! `tracing`:
+//! 
 //! ```
+//! use opentelemetry_semantic_conventions as semconv;
+//! use tracing::span;
```
Comment (jtescher) [inline opentelemetry-semantic-conventions/scripts/templates/header_resource.rs:14]: Does tracing now need to be a dev dep?

**Review by jtescher — CHANGES_REQUESTED** [review]
Overall supportive of this change, some nits left

File: opentelemetry-semantic-conventions/src/semconv/resource.rs (hunk 2 above)
Comment (kriswuollett) [reply opentelemetry-semantic-conventions/src/semconv/resource.rs:1]: Left in by mistake, removing.

File: opentelemetry-semantic-conventions/scripts/templates/header_resource.rs (hunk 4 above)
Comment (kriswuollett) [reply opentelemetry-semantic-conventions/scripts/templates/header_resource.rs:14]: Seems like it should be, adding.

File: opentelemetry-semantic-conventions/scripts/templates/header_resource.rs [hunk 5]
```diff
@@ -7,14 +7,31 @@
...
+//! [`tracing`]:
+//! 
 //! ```
+//! use opentelemetry_semantic_conventions as semconv;
+//! use tracing::span;
+//!
+//! let span = span!(
+//!     LEVEL::INFO,
+//!     "handle_request",
+//!     { semconv::resource::SERVICE_NAME = "my-service" },
```
Comment (cijothomas) [inline opentelemetry-semantic-conventions/scripts/templates/header_resource.rs:19]: I am trying to understand this line here.. Is this adding service name and ns to every span as attributes? If yes, that is not something we should show/encourage, as Resource is expected to be attached to the Provider level *not* to each span/log/metric. (In somebackends, it maybe ultimately stored for each span/log, but thats at exporter level//backend level)

File: opentelemetry-semantic-conventions/scripts/templates/header_resource.rs [hunk 6]
```diff
@@ -7,14 +7,31 @@
 //!
 //! ## Usage
 //!
+//! [`tracing`]:
```
Comment (cijothomas) [inline opentelemetry-semantic-conventions/scripts/templates/header_resource.rs:10]: I think examples in this repo should show OTel Tracing API, not `tracing` crate macros!

[I agree with the decision](https://github.com/open-telemetry/opentelemetry-rust/issues/1043#issuecomment-1776529519) to specially treat `tracing` in the API/SDK given the importance of tracing crate to Rust ecosystem, but the examples in this repo should be showing components shipped from this repo.

**Review by cijothomas — CHANGES_REQUESTED** [review]
Changes looks good! 

I am leaving my review as "request-changes", specifically for addressing the below points:
https://github.com/open-telemetry/opentelemetry-rust/pull/1334/files#r1387150519
https://github.com/open-telemetry/opentelemetry-rust/pull/1334/files#r1387146406

File: opentelemetry-semantic-conventions/scripts/templates/header_resource.rs (hunk 5 above)
Comment (kriswuollett) [reply opentelemetry-semantic-conventions/scripts/templates/header_resource.rs]: OK, just removed.

File: opentelemetry-semantic-conventions/scripts/templates/header_resource.rs (hunk 6 above)
Comment (kriswuollett) [reply opentelemetry-semantic-conventions/scripts/templates/header_resource.rs]: OK, lacking a specific suggested example to do, if any, for `tracing`, I've just removed the additional example block. I'll leave it to any follow ups that anyone would like to do for docs.

**Review by jtescher — APPROVED** [review]

**Pull request merged**